llvm/lib/Transforms/Utils/SSAUpdaterBulk.cpp

```diff
@@ -143,34 +136,65 @@ void SSAUpdaterBulk::RewriteAllUses(DominatorTree *DT,
     SmallVector<BasicBlock *, 32> IDFBlocks;
     SmallPtrSet<BasicBlock *, 32> LiveInBlocks;
     ComputeLiveInBlocks(UsingBlocks, DefBlocks, LiveInBlocks, PredCache);
-    IDF.resetLiveInBlocks();
     IDF.setLiveInBlocks(LiveInBlocks);
     IDF.calculate(IDFBlocks);
 
+    // Reserve sufficient buckets to prevent map growth. [1]
+    BBInfos.init(LiveInBlocks.size() + DefBlocks.size());
```

Comment:
```suggestion
    BBInfos.reserve(LiveInBlocks.size() + DefBlocks.size());
```
I don't think init() is supposed to be a public API, probably just exposed by accident.

Reply:
Thanks, do you believe I shouldn't use it and cannot assume the DenseMap cannot grow if reserved enough?

Reply:
Sorry, I overlooked your suggestion, I'll use reserve, thanks!